libs/datamodel/core/src/ast/mod.rs

```diff
@@ -45,125 +45,65 @@ pub(crate) use renderer::Renderer;
 /// during validation of a schema, especially when implementing custom attributes.
 ///
 /// The AST is not validated, also fields and attributes are not resolved. Every node is
-/// annotated with it's location in the text representation.
+/// annotated with its location in the text representation.
 /// Basically, the AST is an object oriented representation of the datamodel's text.
-/// A prisma schema.
 /// Schema = Datamodel + Generators + Datasources
 #[derive(Debug, Clone, PartialEq)]
 pub struct SchemaAst {
     /// All models, enums, datasources, generators or type aliases
-    pub(crate) tops: Vec<Top>,
+    pub(super) tops: Vec<Top>,
 }
 
 impl SchemaAst {
     pub fn empty() -> Self {
         SchemaAst { tops: Vec::new() }
     }
 
-    pub fn find_source(&self, source: &str) -> Option<&SourceConfig> {
-        self.sources().into_iter().find(|s| s.name.name == source)
-    }
-
-    pub fn find_source_mut(&mut self, source: &str) -> Option<&mut SourceConfig> {
-        self.tops.iter_mut().find_map(|top| match top {
-            Top::Source(source_config) if source_config.name.name == source => Some(source_config),
-            _ => None,
-        })
-    }
-
-    pub fn find_model(&self, model: &str) -> Option<&Model> {
+    pub(crate) fn find_model(&self, model: &str) -> Option<&Model> {
         self.models().into_iter().find(|m| m.name.name == model)
     }
```

Comment:
`find_model()`, `find_type_alias()` and `find_field()` would be a bit too much diff for this PR, so I haven't touched them.